src/Microsoft.Data/BaseColumn.cs

```diff
@@ -65,9 +65,7 @@ protected set
         /// <param name="mapIndices"></param>
         /// <param name="invertMapIndices"></param>
         /// <returns></returns>
-        public virtual BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapIndices = false) => throw new NotImplementedException();
-
-        public virtual BaseColumn CloneAndAppendNulls(BaseColumn mapIndices = null, bool invertMapIndices = false) => throw new NotImplementedException();
+        public virtual BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapIndices = false, long numberOfNullsToAppend = 0) => throw new NotImplementedException();
```

Comment:
Not necessarily needs to be done in this PR, but you may want to consider creating a `CloneOptions` class that holds all these options. That way we can add future options without breaking the API.

Unless you think this is all the options `Clone` will ever need.